### PR TITLE
Cosmetics

### PR DESCRIPTION
**Pull request by bigfluffycookie**

### CODE REVIEW

File: Exercise/Exercise/InputValidator.cs [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
 {
     public class InputValidator
     {
-        public static bool IsValidFilePath(string filePath, string fileExtension)
-        {
-            return FileExists(filePath) && FileHasCorrectExtension(filePath, fileExtension);
-        }
+        private IFileSystem fileSystem;
+
+        public InputValidator(IFileSystem fileSystem) => this.fileSystem = fileSystem;
```
Comment (rita-sonarsource) [inline Exercise/Exercise/InputValidator.cs:9]: Small cosmetics note: for CTORs we keep the {} brackets, and not lamdas. No real reason why except it's the industry standard.

File: Exercise/Exercise/Program.cs [hunk 2]
```diff
@@ -128,7 +130,7 @@ private static string ReadUserInputForFilePath(string fileExtension)
                 }
 
                 filePath = input;
-            } while (!InputValidator.IsValidFilePath(filePath, fileExtension));
+            } while (!(inputValidator.FileExists(filePath) && inputValidator.FileHasCorrectExtension(filePath, fileExtension)));
```
Comment (rita-sonarsource) [inline Exercise/Exercise/Program.cs:133]: I'm curious, why did you remove the previous method? 
It's not right or wrong, just a question of preference.

File: Exercise/Exercise.UnitTests/InputValidatorTests.cs [hunk 3]
```diff
@@ -1,47 +1,64 @@
...
+[TestClass]
+public class InputValidatorTests
 {
-    [TestClass]
-    public class InputValidatorTests
+    [TestMethod]
+    public void FileHasCorrectExtension_WrongExtension_ReturnsFalse()
+    {
+        var filePath = "Test.txt";
+        var fileSystem = new MockFileSystem(new Dictionary<string, MockFileData>());
```
Comment (rita-sonarsource) [inline Exercise/Exercise.UnitTests/InputValidatorTests.cs:13]: The MockFileSystem classes are useful but not necessarily needed.
Since your class uses an interface now, you can treat it as a normal `Mock<IFileSystem>` and do the set up just as you would normally.

The `MockFileSystem` helper classes are useful when you want to mock more complicated file system operations, i.e. Search. Otherwise a mock is good enough and a bit easier to read and use.

File: Exercise/Exercise.UnitTests/InputValidatorTests.cs [hunk 4]
```diff
@@ -1,47 +1,64 @@
...
+
+[TestClass]
+public class InputValidatorTests
 {
-    [TestClass]
-    public class InputValidatorTests
+    [TestMethod]
+    public void FileHasCorrectExtension_WrongExtension_ReturnsFalse()
+    {
+        var filePath = "Test.txt";
```
Comment (rita-sonarsource) [inline Exercise/Exercise.UnitTests/InputValidatorTests.cs:12]: const string

**Review by rita-sonarsource — COMMENTED** [review]
Looks good! see my comments

File: Exercise/Exercise/InputValidator.cs [hunk 5]
```diff
@@ -1,15 +1,18 @@
...
+
+namespace Exercise
 {
     public class InputValidator
     {
-        public static bool IsValidFilePath(string filePath, string fileExtension)
-        {
-            return FileExists(filePath) && FileHasCorrectExtension(filePath, fileExtension);
-        }
+        private IFileSystem fileSystem;
```
Comment (rita-sonarsource) [inline Exercise/Exercise/InputValidator.cs:7]: readonly

File: Exercise/Exercise/Program.cs (hunk 2 above)
Comment (bigfluffycookie) [reply Exercise/Exercise/Program.cs:133]: Mostly because I felt like it was not useful, especially as the name did not really portray what it did and it is clearer to use the methods seperatly

**Review by rita-sonarsource — APPROVED** [review]
